### PR TITLE
Add 64-bit version of geqrf, geqr2

### PR DESCRIPTION
**Pull request by qjojo**
Adds 64-bit API support to geqr2 and geqrf along with their strided and batched variants.

### CODE REVIEW

File: library/src/lapack/roclapack_geqrf.cpp [hunk 1]
```diff
@@ -48,12 +44,12 @@ rocblas_status rocsolver_geqrf_impl(rocblas_handle handle,
         return st;
 
     // working with unshifted arrays
-    rocblas_int shiftA = 0;
+    I shiftA = 0;
```
Comment (tfalders) [inline library/src/lapack/roclapack_geqrf.cpp]: Should be rocblas_stride

**Review by tfalders — CHANGES_REQUESTED** [review]
Looking good so far, but I think you missed geqr2_batched.cpp, geqr2_strided_batched.cpp, geqrf_batched.cpp, and geqrf_ptr_batched.cpp.

EDIT: Also, it looks like you need to resolve a conflict with the PR I merged. Sorry about that.

**Comment by qjojo** [comment]
Fixed some missing exports and added guards to allow building without the 64-bit API.

**Review by tfalders — APPROVED** [review]
Looks good!

**Review by jzuniga-amd — APPROVED** [review]
LGTM

**Review by EdDAzevedo — APPROVED** [review]

**Comment by qjojo** [comment]
Since the failures are known and unrelated I will manually merge this.

**Pull request merged**